### PR TITLE
Add Rust game workshop

### PR DESCRIPTION
**Pull request by panos**
Let's try this again...

CC: @cfanoulis

### CODE REVIEW

**Review by sampoder — APPROVED** [review]
I'm satisfied with this, thank you @panos @cfanoulis for your work on this.

File: workshops/rust_text_game/README.md
Comment (sampoder) [inline workshops/rust_text_game/README.md:5]:
```suggestion
author: "@panos, @cfanoulis"
```

This is just a quick fix, just I can't seem to make edits as an admin.

Could you please enable this? I'd help @MatthewStanciu and I.

Thanks!

Comment (cfanoulis) [reply workshops/rust_text_game/README.md:5]: @panos since I can neither apply changes

**Comment by panos** [comment]
Mind if I squash the commits and sign before merging?

**Comment by panos** [comment]
Or, nevermind, this is going to take some rebasing either way, so it'd be great if you could do that from the GitHub platform, since it'll probably save more time. Do you want me to give you access, just in case there's anything more to correct? @sampoder

**Comment by sampoder** [comment]
We'll click this button to squash them all together when merged. @panos that'd be great could you do it for @MatthewStanciu as well?

Screenshot 2021-01-20 at 9 01 03 PM

**Comment by sampoder** [comment]
This is so annoying: https://github.com/isaacs/github/issues/1681

**Comment by cfanoulis** [comment]
I'll go ahead and rebase-squash-sign stuff in a moment


my gpg keys are here if anyone wants to verify https://keybase.io/cfanoulis

**Comment by panos** [comment]
@cfanoulis Leave it up to me.

@sampoder Adding you to the repo. I didn't quite catch what's up with @MatthewStanciu?

**Comment by sampoder** [comment]
@MatthewStanciu will be doing a final check / review

**Comment by panos** [comment]
Got it, we gave him write access in the case his remarks happen to involve requesting relatively minor changes to our workshop.

**Comment by MatthewStanciu** [comment]
@panos GIBE ME WRITE ACCESS PLEASE I WILL ONLY MKAE SMALL CHANGES I PROMISE UWU

**Comment by cfanoulis** [comment]
give me a second while I run prettier on this kthanks

**Comment by MatthewStanciu** [comment]
Omg this workshop is sooooo good!!! I'm so excited we finally have a beginner Rust workshop. FYI I went in and made some small edits. Here's what I changed:

- There was a blank line before the metadata, which actually broke the metadata. So I moved it up.
- A few code blocks (mostly near the beginning) didn't specify the language, so I added a language
- The formatting in the code blocks was inconsistent—some had 2 spaces, some had 4 spaces, some had a mix of tabs and spaces—so I made them consistent. To optimize for speeeeeeeeed, I took the opinionated approach and just made everything 2 spaces. You're welcome to change it though if you have a different opinion than me
- I made one or two super minimal wording edits

If y'all are okay with the changes I made, let me know. I think it's ready to merge 🚀

**Comment by MatthewStanciu** [comment]
YAY

**Pull request merged**
